src/main/java/io/cdap/plugin/add/field/MultiFieldAdderConfig.java

```diff
@@ -39,18 +39,21 @@ public static Builder builder(MultiFieldAdderConfig copy) {
 
   Map<String, String> getFieldValue() throws IllegalArgumentException {
     Map<String, String> values = new TreeMap<>();
-    if (containsMacro(FIELD_VALUE) || fieldValue == null || fieldValue.trim().isEmpty()) {
+    if (containsMacro(FIELD_VALUE)) {
+      fieldValue = getMacroFieldValue();
+    }
+    if (fieldValue == null || fieldValue.trim().isEmpty()) {
       return values;
     }
 
     String[] fvPairs = fieldValue.split(",");
     for (String fvPair : fvPairs) {
       String[] fieldAndValue = fvPair.split(":");
-      if (fieldAndValue.length != 2) {
+      if (fieldAndValue.length != 2 || fvPair.startsWith("$")) {
         continue;
       }
       String fieldName = fieldAndValue[0];
-      String fieldValue = fieldAndValue[1];
+      String fieldValue = fieldAndValue[1].startsWith("$") ? null : fieldAndValue[1];
```

Comment:
should not assume starts with. macro can be nested. Should be contains

Reply:
How is this being added to the output schema? mind adding screenshot?

Reply:
1. PR updated.
2. Screenshot: 
![image (1)](https://user-images.githubusercontent.com/39129331/109498750-2e91fd00-7a94-11eb-93da-b1f69bb080cf.png)